lib/luaxtable.c

```diff
@@ -137,13 +251,22 @@ static int luaxtable_new##hook(lua_State *L) 				\
 	luaxtable_setinteger(L, 1, hook, revision);			\
 	luaxtable_setinteger(L, 1, hook, family);			\
 	luaxtable_setinteger(L, 1, hook, proto);			\
+	lunatik_checkfield(L, 1, "checkentry", LUA_TFUNCTION);		\
+	lunatik_checkfield(L, 1, "destroy", LUA_TFUNCTION);		\
+	lunatik_checkfield(L, 1, #hook, LUA_TFUNCTION);			\
 									\
+	hook->usersize = 0;						\
+	hook->hook##size = sizeof(luaxtable_info_t);			\
 	hook->hook = luaxtable_##hook;					\
 	hook->checkentry = luaxtable_##hook##_check;			\
 	hook->destroy = luaxtable_##hook##_destroy;			\
 									\
+	if (luarcu_settable(luaxtable_hooks.hook, hook->name, XT_EXTENSION_MAXNAMELEN, object) != 0)	\
+		luaL_error(L, "unable to set key : %s\n", hook->name);	\
```

Comment:
I think we might have a better error message to the API user.. `unable to set key` is a bit cryptic.. which key? key for what? perhaps we just need to signalize that we were `unable to create the hook` (or something like that)